test/drivers/test_rsvd.cc

```diff
@@ -17,91 +17,118 @@ class TestRSVD : public ::testing::Test
-        std::vector<T> A_hat(size, 0.0);
-        std::vector<T> A_hat_buf (m * k, 0.0);
-        std::vector<T> A_k(size, 0.0);
-        std::vector<T> A_cpy (m * n, 0.0);
+        std::vector<T> A_hat;
+        std::vector<T> A_hat_buf;
+        std::vector<T> A_k;
+        std::vector<T> A_cpy;
 
         // For RSVD
-        std::vector<T> s1(n, 0.0);
-        std::vector<T> S1(n * n, 0.0);
-        std::vector<T> U1(m * n, 0.0);
-        std::vector<T> VT1(n * n, 0.0);
+        std::vector<T> s1;
+        std::vector<T> S1;
+        std::vector<T> U1;
+        std::vector<T> VT1;
 
         // For low-rank SVD
-        std::vector<T> s(n, 0.0);
-        std::vector<T> S(n * n, 0.0);
-        std::vector<T> U(m * n, 0.0);
-        std::vector<T> VT(n * n, 0.0);
-
-        T* A_dat = A.data();
+        std::vector<T> s;
+        std::vector<T> S;
+        std::vector<T> U;
+        std::vector<T> VT;
 
-        T* A_hat_dat = A_hat.data();
-        T* A_hat_buf_dat = A_hat_buf.data();
-        T* A_k_dat = A_k.data();
-        T* A_cpy_dat = A_cpy.data();
+        RSVDTestData(int64_t m, int64_t n, int64_t k) :
+        A(m * n, 0.0),
+        // For results comparison
+        A_hat(m * n, 0.0),
+        A_hat_buf (m * k, 0.0),
+        A_k(m * n, 0.0),
+        A_cpy (m * n, 0.0),
 
-        T* U1_dat = U1.data();
-        T* S1_dat = S1.data();
-        T* VT1_dat = VT1.data();
+        // For RSVD
+        s1(n, 0.0),
+        S1(n * n, 0.0),
+        U1(m * n, 0.0),
+        VT1(n * n, 0.0),
 
-        T* U_dat = U.data();
-        T* s_dat = s.data();
-        T* S_dat = S.data();
-        T* VT_dat = VT.data();
+        // For low-rank SVD
+        s(n, 0.0),
+        S(n * n, 0.0),
+        U(m * n, 0.0),
+        VT(n * n, 0.0)
+        {}
+    };
 
+    template <typename T, typename RNG>
+    struct algorithm_objects {
+        RandLAPACK::PLUL<T> Stab;
+        RandLAPACK::RS<T, RNG> RS;
+        RandLAPACK::CholQRQ<T> Orth_RF;
+        RandLAPACK::RF<T> RF;
+        RandLAPACK::CholQRQ<T> Orth_QB;
+        RandLAPACK::QB<T> QB;
+        RandLAPACK::RSVD<T> RSVD;
+
+        algorithm_objects(bool verbosity, 
+                            bool cond_check, 
+                            bool orth_check, 
+                            int64_t p, 
+                            int64_t passes_per_iteration, 
+                            int64_t block_sz,
+                            RandBLAS::base::RNGState<RNG> state) :
+                            Stab(cond_check, verbosity),
+                            RS(Stab, state, p, passes_per_iteration, verbosity, cond_check),
+                            Orth_RF(cond_check, verbosity),
+                            RF(RS, Orth_RF, verbosity, cond_check),
+                            Orth_QB(cond_check, verbosity),
+                            QB(RF, Orth_QB, verbosity, orth_check),
+                            RSVD(QB, verbosity, block_sz)
+                            {}
```

Comment:
Please format this in the way that I suggested for REVD2.

Reply:
Fixed.